### PR TITLE
PXC-4163: pxc_strict_mode does not detect version mismatch

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://jira.percona.com/browse/PXC-4163

This PR also fixes the following issues:
PXC-4162: wsrep_cluster_size at 0 when syncing from earlier version https://jira.percona.com/browse/PXC-4162

PXC-3976: wsrep status vars are not updated when 8.0 node joins the 5.7 cluster
https://jira.percona.com/browse/PXC-3976

Problem:
PXC-4163:
During the rolling upgrade, pxc_strict_mode = ENFORCING does not prevent DDL and DML operations on 8.0.x node if there is 5.7.x node present in the cluster.

PXC-4162 and PXC-3976:
During the rolling upgrade, wsrep-related status variables are not properly set/updated on 8.0 nodes.

Cause:
1. Galera 4.x does not notify the application about the current cluster view via view_cb in case state transfer is requested. It notifies application via connected_cb, but application makes no use of this.
2. 5.7 does not generate new seqno for CC, so if we are in the mixed version cluster (5.7 and 8.0 nodes) and nodes are: node_1 (5.7), node_2 (8.0 - this node), node_3 (8.0 - just joining) and there was no any DDL/DML between node_2 and node_3 join, during node_3 join we still see the same conf.seqno as it was recorded during node_2's SST and notification about the view change is skipped.

Solution:
Added missing cluster view notification.

### CODE REVIEW

File: galera/src/replicator_smm.cpp [hunk 1]
```diff
@@ -3158,9 +3159,18 @@ void galera::ReplicatorSMM::process_prim_conf_change(void* recv_ctx,
...
+
+        /* 5.7 does not generate new seqno for CC, so if we are in the
+           mixed version cluster (5.7 and 8.0 nodes) and nodes are:
+           node_1 (5.7), node_2 (8.0 - this node), node_3 (8.0 - just joining)
+           and there was no any DDL/DML between node_2 and node_3 join,
+           during node_3 join we still see the same conf.seqno as it was
+           recorded during node_2's SST. */
+        if (conf.seqno == sst_seqno_ && mixed_ver_cluster) {
+            submit_ordered_view_info(recv_ctx, view_info.get());
+        }
```
Comment (venkatesh-prasad-v) [inline galera/src/replicator_smm.cpp]: Here and in the next chunk. Please use galera style formatting.
```suggestion
        if (conf.seqno == sst_seqno_ && mixed_ver_cluster) 
        {
            submit_ordered_view_info(recv_ctx, view_info.get());
        }
```

File: galera/src/replicator_smm.cpp [hunk 2]
```diff
@@ -3150,6 +3150,7 @@ void galera::ReplicatorSMM::process_prim_conf_change(void* recv_ctx,
     /* Invalidate sst_seqno_ in case of group change. */
     if (state_uuid_ != group_uuid) sst_seqno_ = WSREP_SEQNO_UNDEFINED;
 
+    bool mixed_ver_cluster = view_info->proto_ver < 4;
```
Comment (venkatesh-prasad-v) [inline galera/src/replicator_smm.cpp]: Use `const` similar to all local var initializations in this function.
```suggestion
    bool const mixed_ver_cluster = view_info->proto_ver < 4;
```

File: galera/src/replicator_smm.cpp (hunk 1 above)
Comment (kamil-holubicki) [reply galera/src/replicator_smm.cpp]: done

File: galera/src/replicator_smm.cpp (hunk 2 above)
Comment (kamil-holubicki) [reply galera/src/replicator_smm.cpp]: done

**Comment by kamil-holubicki** [comment]
https://pxc.cd.percona.com/job/pxc-8.0-pipeline-parallel-mtr/1413/

**Review by venkatesh-prasad-v — APPROVED** [review]

**Pull request merged**